src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolution.cs

```diff
@@ -615,7 +616,7 @@ public static bool IsValidParams(Symbol member)
 
             // Note: we need to confirm the "arrayness" on the original definition because
             // it's possible that the type becomes an array as a result of substitution.
-            ParameterSymbol final = member.GetParameters().Last();
+            var final = member.GetParameters().Last();
```

Comment:
Rolled back the change, this was changed because I initially made changes in this area to try a few scenarios.